pkg/apis/harvesterhci.io/v1beta1/addon.go

```diff
@@ -1,15 +1,31 @@
 package v1beta1
 
 import (
+	"github.com/rancher/wrangler/pkg/condition"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 type AddonState string
 
 const (
-	AddonEnabled  AddonState = "AddonEnabled"
-	AddonDeployed AddonState = "AddonDeploySuccessful"
-	AddonFailed   AddonState = "AddonDeployFailed"
+	AddonEnabling  AddonState = "AddonEnabling"
+	AddonDeployed  AddonState = "AddonDeploySuccessful"
+	AddonDisabled  AddonState = "AddonDisabled"
+	AddonDisabling AddonState = "AddonDisabling"
+
+	// after successfully updating, addon will be AddonInitState when !Spec.Enaabled; AddonDeployed when Spec.Enaabled
```

Comment:
```suggestion
	// after successfully updating, addon will be AddonInitState when !Spec.Enabled; AddonDeployed when Spec.Enabled
```
nit: typo in comment.